app/src/features/sources/SourceForm.tsx

```diff
@@ -17,15 +18,71 @@ import {
   useCreateSourceMutation,
   useUpdateSourceMutation,
 } from "services/api/api";
-import { Source } from "services/api/generated/api.generated";
+import { Source, Credential } from "services/api/generated/api.generated";
 
 import { editSource, selectSourceToEdit } from "./sourceSlice";
 
-const inputs: (t: TFunction) => FormInputProperty<Source>[] = (t) => [
+const credentialInputs: (t: TFunction) => FormInputProperty<Credential>[] = (
+  t
+) => [
   {
     type: "text",
-    name: "name",
-    label: t("sourceName"),
+    name: "host",
+    label: t("host"),
+    variant: "outlined",
+    validationRules: { required: true },
+  },
+  {
+    type: "number",
+    name: "port",
+    label: t("port"),
+    variant: "outlined",
+    validationRules: { required: true },
```

Comment:
It is possible to set rules over max/min values like so 
```suggestion
    validationRules: { 
         required: true,
         min: { value: 1, message: t("noNegativeValue") },
    },
```